pandas/core/arrays/arrow/array.py

```diff
@@ -57,6 +57,34 @@
         "ge": pc.greater_equal,
     }
 
+    ARROW_LOGICAL_FUNCS = {
+        "and": pc.and_kleene,
+        "rand": lambda x, y: pc.and_kleene(y, x),
+        "or": pc.or_kleene,
+        "ror": lambda x, y: pc.or_kleene(y, x),
+        "xor": pc.xor,
+        "rxor": lambda x, y: pc.xor(y, x),
+    }
+
+    ARROW_ARITHMETIC_FUNCS = {
+        "add": pc.add_checked,
+        "radd": lambda x, y: pc.add(y, x),
+        "sub": pc.subtract_checked,
+        "rsub": lambda x, y: pc.subtract_checked(y, x),
+        "mul": pc.multiply_checked,
+        "rmul": lambda x, y: pc.multiply_checked(y, x),
+        "truediv": NotImplemented,  # pc.divide_checked,
```

Comment:
@jorisvandenbossche is it expected for `divide_checked` to have different semantics for integers?

```
In [1]: import pyarrow as pa

In [2]: 5 / 2
Out[2]: 2.5

In [3]: np.int64(5) / 2
Out[3]: 2.5

In [4]: pa.compute.divide_checked(pa.scalar(5), 2)
Out[4]: <pyarrow.Int64Scalar: 2>

In [5]: 5. / 2
Out[5]: 2.5

In [6]: np.float64(5) / 2
Out[6]: 2.5

In [7]: pa.compute.divide_checked(pa.scalar(5.), 2)
Out[7]: <pyarrow.DoubleScalar: 2.5>

In [8]: pa.__version__
Out[8]: '7.0.0
```

Reply:
The division kernels are only implemented for int input -> int output, float input -> float output, etc. Since in our case division with integers always results in floats, I think we can emulate that with casting the integer input to float first (that's for example also what the arrow R dplyr bindings do).

Reply:
Thanks for confirming the behavior!

I think this touches upon a larger philosophical question for "using arrow in pandas"

1. Should pandas make arrow follow pandas' conventions? (e.g. pandas casts such that int / int -> float)
2. Should pandas just be an interface for arrow to propagate its conventions? (e.g. int / int -> int)

Probably should be discussed with a larger audience, but personally I've been leaning 2.

Reply:
In practice I suppose it will be something in the middle? 

We certainly shouldn't blindly follow existing pandas behaviours, but I think we will also certainly need to add some user convenience layer on top of Arrow (now, in this specific case, it's of course more an API behaviour choice rather than a user convenience issue).

(Arrow needs to cater for many different bindings and use cases (databases, dataframe libraries), so sometimes will also have more options / different defaults / different names than what a Python user expects). 

But indeed something interesting to discuss (both more broadly in general terms, and we will probably still often need some discussion on a case by case basis).